kedro/pipeline/node.py

```diff
@@ -128,6 +128,10 @@ def __init__(  # noqa: PLR0913
 
         self._func = func
         self._inputs = inputs
+        # Cache the list version of inputs dictionary to avoid recomputing in the inputs property
+        self._dict_inputs_as_list = None
+        if isinstance(self._inputs, dict):
+            self._dict_inputs_as_list = _dict_inputs_to_list(self._func, self._inputs)
```

Comment:
```suggestion
        self._inputs = _dict_inputs_to_list(self._func, self._inputs) if isinstance(inputs, dict) else inputs
```
Any reason this wouldn't just work?